crates/sui-graphql-rpc/src/types/move_enum.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use sui_package_resolver::{EnumDef, VariantDef};
+
+use crate::error::Error;
+
+use super::{
+    move_module::MoveModule,
+    move_struct::{MoveField, MoveStructTypeParameter},
+    open_move_type::{abilities, MoveAbility},
+    sui_address::SuiAddress,
+};
+
+pub(crate) struct MoveEnum {
+    defining_id: SuiAddress,
+    module: String,
+    name: String,
+    abilities: Vec<MoveAbility>,
+    type_parameters: Vec<MoveStructTypeParameter>,
+    variants: Vec<MoveEnumVariant>,
+    checkpoint_viewed_at: u64,
+}
+
+/// Information for a particular Move variant
+pub(crate) struct MoveEnumVariant {
+    name: String,
+    fields: Vec<MoveField>,
+}
+
+/// Description of an enum type, defined in a Move module.
+#[Object]
+impl MoveEnum {
+    /// The module this enum was originally defined in.
+    pub(crate) async fn module(&self, ctx: &Context<'_>) -> Result<MoveModule> {
+        let Some(module) = MoveModule::query(
+            ctx.data_unchecked(),
+            self.defining_id,
+            &self.module,
+            self.checkpoint_viewed_at,
+        )
+        .await
+        .extend()?
+        else {
+            return Err(Error::Internal(format!(
+                "Failed to load module for enum: {}::{}::{}",
+                self.defining_id, self.module, self.name,
+            )))
+            .extend();
+        };
+
+        Ok(module)
+    }
+
+    /// The enums's (unqualified) type name.
+    pub(crate) async fn name(&self) -> &str {
+        &self.name
+    }
+
+    /// Abilities this enum has.
```

Comment:
```suggestion
    /// The enum's abilities.
```

to mirror the doc comment above maybe